src/logion-chain/Vault.ts

```diff
@@ -0,0 +1,42 @@
+import { keyring } from "@polkadot/ui-keyring";
+import { RecoveryConfig } from "./Recovery";
+
+const THRESHOLD = 2;
+const NAME = "LogionVault";
+
+export function getVault(address: string, recoveryConfig: RecoveryConfig): string {
+    checkKeyringLoaded()
+    const vaultAddress = findVault();
+    return vaultAddress ? vaultAddress : createVault(address, recoveryConfig)
+}
+
+function createVault(address: string, recoveryConfig: RecoveryConfig): string {
```

Comment:
The name of this function is a bit confusing as do not actually create something. What about `addVaultToKeyring`?

Reply:
Agreed.